app/org/sagebionetworks/bridge/services/UploadService.java

```diff
@@ -236,12 +259,10 @@ public Upload getUpload(@Nonnull String uploadId) {
             throw new BadRequestException("Query window cannot be longer than two days: " + startTime + "-" + endTime);
         }
         
-        List<? extends Upload> results = uploadDao.getUploads(healthCode, startTime, endTime);
-        
-        List<UploadView> views = results.stream().map(upload -> {
+        List<UploadView> views = supplier.get(startTime, endTime).stream().map(upload -> {
             UploadView.Builder builder = new UploadView.Builder();
             builder.withUpload(upload);
-            if (upload.getStatus() == UploadStatus.SUCCEEDED) {
+            if (includeSchemaInfo && upload.getStatus() == UploadStatus.SUCCEEDED) {
                 UploadValidationStatus status = getUploadValidationStatus(upload.getUploadId());
```

Comment:
Instead of calling getUploadValidationStatus(), you could call getRecordById() in either HealthDataService or HealthDataDao. This will save you the redundant call to getUploads(). You can also key on the presence or absence of recordId, which is equivalent to UploadStatus.SUCCEEDED, but more direct.